zip-0316.rst

```diff
@@ -379,24 +379,25 @@ will normally take the union of information provided by all contained
 Receivers, and therefore the Priority List defined in the previous
 section is not used.
 
-For each UFVK Type or UIVK Type currently defined in this specification,
+For each FVK Type or IVK Type currently defined in this specification,
 the same Typecode is used as for the corresponding Receiver Type in a
-Unified Address. Additional UFVK Types and UIVK Types MAY be defined in
+Unified Address. Additional FVK Types and IVK Types MAY be defined in
 future, and these will not necessarily use the same Typecode as the
 corresponding Unified Address.
 
-The following UFVK or UIVK Encodings are used in place of the
+The following FVK or IVK Encodings are used in place of the
 :math:`\mathtt{addr}` field:
 
-* An Orchard UFVK or UIVK Encoding, with Typecode :math:`\mathtt{0x03},`
+* An Orchard FVK or IVK Encoding, with Typecode :math:`\mathtt{0x03},` is
   is the raw encoding of the Orchard Full Viewing Key or Orchard Incoming
   Viewing Key respectively.
 
-* A Sapling UFVK Encoding, with Typecode :math:`\mathtt{0x02},` is
+* A Sapling FVK Encoding, with Typecode :math:`\mathtt{0x02},` is
   the encoding of a Sapling Extended Full Viewing Key defined in
-  [#zip-0032-sapling-extfvk]_.
+  [#zip-0032-sapling-extfvk]_. This SHOULD be an Extended Full Viewing
+  Key at the Account level of the ZIP 32 hierarchy.
```

Comment:
In ZIP sync today, we agreed to change this to instead encode Sapling FVKs using the `EncodeExtFVKParts` function from ZIP 32, so they are a representation of `(ak, nk, ovk, dk)`. This reduces the size of the encoding by 25%, and removes capabilities (`c`) and potentially-private information (child index which is the account number) from the FVK encoding.

The downside is that we can't extract a Sapling ExtFVK from a UFVK losslessly, but that doesn't matter in most cases. If someone needs to do that, they should set `(depth, parent_fvk_tag, i, c)` to the string of all 1 bits, to indicate "data unknown".